compiler_gym/datasets/datasets.py

```diff
@@ -0,0 +1,480 @@
+        self.random = np.random.default_rng(seed)
+        for dataset in self._datasets.values():
+            dataset.seed(seed)
+
+    def datasets(
+        self, inactive: bool = False, hidden: bool = False
+    ) -> Iterable[Dataset]:
+        """Enumerate the datasets.
+
+        Dataset order is consistent across runs.
+
+        :param inactive: If :code:`False` (the default), only datasets that are
+            active will be returned.
+
+        :param hidden: If :code:`False` (the default), only datasets whose
+            :meth:`Dataset.hidden <compiler_gym.datasets.Dataset.hidden>` value
+            is :code:`False` are returned.
+
+        :return: An iterable sequence of :meth:`Dataset
+            <compiler_gym.datasets.Dataset>` instances.
+        """
+        datasets = self._datasets.values()
+        if not inactive:
+            datasets = (d for d in datasets if d.name in self._active_datasets)
+        if not hidden:
+            datasets = (d for d in datasets if not d.hidden)
+        yield from sorted(datasets, key=lambda d: (d.sort_order, d.name))
+
+    def __iter__(self) -> Iterable[Dataset]:
+        """Iterate over the active datasets.
+
+        Dataset order is consistent across runs.
+
+        Equivalent to iterating over :meth:`datasets.datasets()
+        <compiler_gym.datasets.Dataset.datasets>`, but without the ability to
+        iterate over the inactive or hidden datasets.
+
+        :return: An iterable sequence of :meth:`Dataset
+            <compiler_gym.datasets.Dataset>` instances.
+        """
+        yield from self.datasets()
+
+    def dataset(self, dataset: Optional[Union[str, Dataset]] = None) -> Dataset:
+        """Get a dataset.
+
+        If a name is given, return the corresponding :meth:`Dataset
+        <compiler_gym.datasets.Dataset>`. Else, return a dataset uniformly
+        randomly from the set of available datasets.
+
+        Use :meth:`seed() <compiler_gym.datasets.Dataset.seed>` to force a
+        reproducible order for randomly selected datasets.
+
+        Name lookup will succeed whether or not the dataset is active or hidden.
+
+        :param name: A dataset name, or :code:`None` to select a dataset
+            randomly.
+
+        :return: A :meth:`Dataset <compiler_gym.datasets.Dataset>` instance.
+
+        :raises LookupError: If :code:`name` is not found.
+        """
+        if dataset is None:
+            if not self._active_datasets:
+                raise ValueError("No active datasets available")
+
+            return self._datasets[self.random.choice(list(self._active_datasets))]
+
+        if isinstance(dataset, Benchmark):
+            dataset_name = dataset.name
+        else:
+            dataset_name = resolve_uri_protocol(dataset)
+
+        if dataset_name not in self._datasets:
+            raise LookupError(f"Dataset not found: {dataset_name}")
+        return self._datasets[dataset_name]
+
+    def __getitem__(self, name: str) -> Dataset:
```

Comment:
Why doesn't this match the prototype of `dataset()`?

Reply:
Thanks, fixed